applications/OptimizationApplication/python_scripts/controls/thickness/shell_thickness_control.py

```diff
@@ -205,53 +144,53 @@ def Update(self, new_control_field: ContainerExpressionTypes) -> bool:
         if not IsSameContainerExpression(new_control_field, self.GetEmptyField()):
             raise RuntimeError(f"Updates for the required element container not found for control \"{self.GetName()}\". [ required model part name: {self.model_part.FullName()}, given model part name: {new_control_field.GetModelPart().FullName()} ]")
 
-        if Kratos.Expression.Utils.NormL2(self.control_field - new_control_field) > 1e-15:
+        update = new_control_field - self.control_field
+        if Kratos.Expression.Utils.NormL2(update) > 1e-15:
             with TimeLogger(self.__class__.__name__, f"Updating {self.GetName()}...", f"Finished updating of {self.GetName()}.",False):
                 # update the control thickness field
                 self.control_field = new_control_field
                 # now update the physical field
-                self._UpdateAndOutputFields()
+                self._UpdateAndOutputFields(update)
+
+                if self.beta_adaptive:
```

Comment:
Can you please wrap this if block to function __UpdateBeta() ?? I know it is not your implementation, but I hate long spaghetti codes...

Reply:
done :)